storage/innobase/trx/trx0purge.cc

```diff
@@ -707,26 +707,20 @@ TRANSACTIONAL_TARGET void trx_purge_truncate_history()
         ut_ad(!bpage->is_io_fixed());
         ut_ad(bpage->id().space() == space_id);
 
-        if (bpage->oldest_modification() > 2)
-        {
+        if (bpage->oldest_modification() > 1 &&
+            !mtr.have_x_latch(*reinterpret_cast<buf_block_t*>(bpage)))
```

Comment:
I actually intended to retain the `> 2` to be consistent with the outer `if` inside the `for` body, but I forgot to save that change before creating this pull request. Either value is fine (the value 2 is specific to dirty pages of the temporary tablespace), but using the same value could allow the compiler to reuse a register for the common subexpression. In the 10.5 version #2830 we use the value 1 in both.